src/Polly.Core/Strategy/Outcome.cs

```diff
@@ -76,18 +76,24 @@ public bool TryGetResult(out object? result)
     }
 
     /// <summary>
-    /// Gets the string representation of the outcome.
+    /// Returns the string representation of the outcome.
     /// </summary>
-    /// <returns>A string representation of the outcome.</returns>
-    /// <remarks>
-    /// If the outcome represents an exception, then <see cref="Exception.Message"/> will be returned.
-    /// If the outcome represents a result, then <see cref="Result"/> formatted as string will be returned.
-    /// </remarks>
+    /// <returns>
+    /// The exception message if the outcome is an exception; otherwise, the string representation of the result.
+    /// </returns>
     public override string ToString() => ExceptionDispatchInfo != null
         ? Exception!.Message
         : Result?.ToString() ?? string.Empty;
 
-    internal Outcome<TResult> AsOutcome<TResult>() => ExceptionDispatchInfo != null
-        ? new Outcome<TResult>(ExceptionDispatchInfo)
-        : new Outcome<TResult>((TResult)Result!);
+    internal TResult GetResultOrRethrow()
+    {
+        ExceptionDispatchInfo?.Throw();
+        return Result!;
+    }
+
+    internal Outcome<object> AsObjectOutcome() => Convert<object>();
+
+    internal Outcome<T> Convert<T>() => (ExceptionDispatchInfo != null)
```

Comment:
I'm not sure what we should call it, but Convert doesn't seem the right name to me for this. It seems to suggest it turns it into a TResult.

Reply:
Maybe just `AsOutcome` and `AsOutcome<T>`?

Reply:
That seems reasonable to me.